File: DRIVER_MANAGEMENT_PROJECT_BACKEND/src/main/java/com/eren/DRIVER_MANAGEMENT_PROJECT_BACKEND/entity/enums/CadreTypes.java
package com.eren.DRIVER_MANAGEMENT_PROJECT_BACKEND.entity.enums;

/**
 * Represents various cadre types for transport-related tasks.
 *
 * <p>This enumeration defines the categories of transportation services,
 * such as public transport, intercity transport, corporate shuttle services,
 * and other specialized operations.</p>
 */
public enum CadreTypes {

    /** Urban public transportation services (e.g. city buses). */
    URBAN_PUBLIC_TRANSPORT,
    /** Intercity transportation services (e.g. between cities). */
    INTERCITY_TRANSPORT,
    /** Shuttle services for corporate or institutional personnel. */
    CORPORATE_SHUTTLE_SERVICE,
    /** Tourism and charter-based transportation services. */
    TOURISM_AND_CHARTER_SERVICE,
    /** Transportation services during night shifts or overnight operations. */
    NIGHT_SHIFT_OPERATION,
    /** Staff assigned as backup or to fill in for absent personnel. */
    BACKUP_AND_RELIEF_STAFF,
    /** Staff used for training, testing, or performance evaluation. */
    TRAINING_AND_EVALUATION,
    /** VIP or private transport assignments, typically non-public use. */
    VIP_AND_PRIVATE_TRANSPORT,
    /** Transportation service specifically for municipal employees. */
    MUNICIPAL_STAFF_TRANSPORT,
    /** Staff working on airport transfer and logistics services. */
    AIRPORT_LOGISTIC_TRANSFER
}

File: DRIVERI_MANAGEMENT_PROJECT_BACKEND/Entities/Enums/CadreTypes.cs
using System.ComponentModel.DataAnnotations;
namespace Entities.Enums
{
    public enum CadreTypes
    {
        [Display(Name = "Urban Public Transport")]
        URBAN_PUBLIC_TRANSPORT = 0,
        [Display(Name = "Intercity Transport")]
        INTERCITY_TRANSPORT = 1,
        [Display(Name = "Corporate Shuttle Service")]
        CORPORATE_SHUTTLE_SERVICE = 2,
        [Display(Name = "Tourism and Charter Service")]
        TOURISM_AND_CHARTER_SERVICE = 3,
        [Display(Name = "Night Shift Operation")]
        NIGHT_SHIFT_OPERATION = 4,
        [Display(Name = "Backup and Relief Staff Transport")]
        BACKUP_AND_RELIEF_STAFF = 5,
        [Display(Name = "Training and Evaluation Transport")]
        TRAINING_AND_EVALUATION = 6,
        [Display(Name = "VIP and Private Transport")]
        VIP_AND_PRIVATE_TRANSPORT = 7,
        [Display(Name ="Municipal Staff Transport")]
        MUNICIPAL_STAFF_TRANSPORT = 8,
        [Display(Name = "Aiport Logistic Transport")]
        AIRPORT_LOGISTIC_TRANSFER = 9
    }
}
